aspen/http/resource.py

```diff
@@ -7,25 +8,18 @@ class Static(object):
     """
 
     def __init__(self, request_processor, fspath, raw, media_type):
-        self.request_processor = request_processor
+        assert type(raw) is bytes  # sanity check
         self.raw = raw
         self.media_type = media_type
         if media_type == 'application/json':
-            self.media_type = self.request_processor.media_type_json
+            self.media_type = request_processor.media_type_json
+        if self.media_type.startswith('text/'):  # XXX this is bad
```

Comment:
Hrm. Is the benefit of validating worth the cost? I've been thinking we should move in the direction of _less_ work for static files. E.g., back when we were more of an application server I think we were looking at using sendfile. On the other hand, I think we're finding pretty good patterns for optimizing static assets from an Aspen application in production, so maybe we don't care about performance here (since we'll never be able to approach production performance requirements).